src/core/execute.c

```diff
@@ -846,6 +846,12 @@ static int setup_pam(
                         goto fail;
         }
 
+        STRV_FOREACH(e, *env) {
+                pam_code = pam_putenv(handle, *e);
+                if (pam_code != PAM_SUCCESS)
+                        goto fail;
```

Comment:
Hm, why do you use `e` here?
`if (pam_code != PAM_SUCCESS)` we try to `strv_free(e)`. And
```
                Stack trace of thread 234:
                #0  0x00007f0b907b6e55 raise (libc.so.6)
                #1  0x00007f0b907b8a5a abort (libc.so.6)
                #2  0x00007f0b907f7dc0 __libc_message (libc.so.6)
                #3  0x00007f0b9080056a _int_free (libc.so.6)
                #4  0x00007f0b90803acc __libc_free (libc.so.6)
                #5  0x00007f0b91cd04be strv_env_clean_with_callback (libsystemd-shared-230.so)
                #6  0x000056530e500987 exec_child (systemd)
                #7  0x000056530e5010a8 exec_spawn (systemd)
                #8  0x000056530e4e08d2 service_spawn (systemd)
                #9  0x000056530e4e1e89 service_enter_start (systemd)
                #10 0x000056530e4e2170 service_enter_start_pre (systemd)
                #11 0x000056530e4e2f73 service_start (systemd)
                #12 0x000056530e53e53a unit_start (systemd)
                #13 0x000056530e5498a1 job_perform_on_unit (systemd)
                #14 0x000056530e549be2 job_run_and_invalidate (systemd)
                #15 0x000056530e4bf5b1 manager_dispatch_run_queue (systemd)
                #16 0x00007f0b91d92784 source_dispatch (libsystemd-shared-230.so)
                #17 0x00007f0b91d9395e sd_event_dispatch (libsystemd-shared-230.so)
                #18 0x00007f0b91d93d93 sd_event_run (libsystemd-shared-230.so)
                #19 0x000056530e4c2116 manager_loop (systemd)
                #20 0x000056530e4b8aab main (systemd)
                #21 0x00007f0b907a2731 __libc_start_main (libc.so.6)
                #22 0x000056530e4b0a19 _start (systemd)
```